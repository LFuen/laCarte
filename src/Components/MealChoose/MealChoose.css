
.choices {
    display: flex;
    justify-content: space-evenly;
    padding: 1rem 0.5rem;
    font-weight: bold;
    font-size: 20px;
    font-family: "Special Elite", cursive;
    text-decoration: none;
    color: #000000;
    text-align: center;
    margin-top: 50px;
    margin-bottom: 100px;
}


.option a{
    display: flex;
    text-align: center;
    justify-content: space-evenly;
    flex-direction: column;
}

h2{
    text-decoration: underline !important;
    color:rgb(233, 111, 111);
    text-shadow: 2px 2px rgb(82, 82, 82);
}

.option h3{
    display: block;
    font-size: 25px;
    margin-block-start: 20px;
    margin-block-end: 10px;
    margin-inline-start: 0px;
    margin-inline-end: 0px;
    font-weight: bold;
}

.option a:hover{
    width: auto;
}