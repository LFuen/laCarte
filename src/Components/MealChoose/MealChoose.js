import React, {Component} from 'react'
import LaCarteContext from '../../context/LaCarteContext'
import {Link} from 'react-router-dom'
import './MealChoose.css'

class cuisineChoose extends Component{

    static contextType = LaCarteContext

    render(){

        const {cuisines, chefs} = this.context

        if(cuisines)
        return(
            <div className='choices'>
                
                <div className='option'>                
                    <h2>Cuisine Origins :</h2>
                    {cuisines.map(cuisine => {return(
                        <Link to={`/cuisines/${cuisine.id}/${cuisine.origin}`} key={cuisine.id}>
                            <h3>{cuisine.origin}</h3>
                        </Link>
                    )})}               
                </div>

                <div className='option'>
                    <span title='Dinner is Served!'><img className='mainImage' alt='served' src='https://lfuen.github.io/laCarte/images/served.png'/></span> 
                    <Link to='/MealList'><h3>All Meals</h3></Link>
                </div>

                <div className='option'>
                    <h2>Chef :</h2>
                    {chefs.map(chef => {return(
                        <Link to={`/chefs/${chef.chef_name}`} key={chef.id}><h3>{chef.chef_name}</h3></Link>
                    )})}
                </div>
                
            </div>
        )
    }
}



export default cuisineChoose