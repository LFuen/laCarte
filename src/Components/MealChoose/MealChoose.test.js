import React from 'react'
import ReactDOM from 'react-dom'
import renderer from 'react-test-renderer'
import ShallowRenderer from 'react-test-renderer/shallow'
import {BrowserRouter} from 'react-router-dom'
import LaCarteContext from '../../context/LaCarteContext'
import MealChoose from './MealChoose'



describe(`MealChoose Component`, () => {
    describe(`Snapshot`, () => {
        it(`renders the UI as expected`, () => {
            const choice = renderer.create(<BrowserRouter><MealChoose/></BrowserRouter>)
                expect(choice).toMatchSnapshot()
        })
    })

    describe(`Smoke test`, () => {
        it(`Renders without crashing`, () => {
            
            const div = document.createElement('div');
            const renderer = new ShallowRenderer()

            const value = {
                meals: 'string',
                chefs: 'string',
                cuisines: 'string',
                orders: 'string',
                addOrder: 'string',
                updateOrder: 'string',
                deleteOrder: 'string',
            };

            renderer.render(
                <BrowserRouter>
                    <LaCarteContext.Provider value={value}>
                        <MealChoose />
                    </LaCarteContext.Provider>                
                </BrowserRouter>,
                div)
            ReactDOM.unmountComponentAtNode(div)
        })
    })
})