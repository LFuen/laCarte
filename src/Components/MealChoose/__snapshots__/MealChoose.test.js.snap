// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MealChoose Component Snapshot renders the UI as expected 1`] = `
<div
  className="choices"
>
  <div
    className="option"
  >
    <h2>
      Cuisine Origins :
    </h2>
  </div>
  <div
    className="option"
  >
    <span
      title="Dinner is Served!"
    >
      <img
        alt="served"
        className="mainImage"
        src="https://lfuen.github.io/laCarte/images/served.png"
      />
    </span>
    <a
      href="/MealList"
      onClick={[Function]}
    >
      <h3>
        All Meals
      </h3>
    </a>
  </div>
  <div
    className="option"
  >
    <h2>
      Chef :
    </h2>
  </div>
</div>
`;
